drivers/gpio/gpio_mcux_lpc.c

```diff
@@ -374,6 +374,9 @@ static const clock_ip_name_t gpio_clock_names[] = GPIO_CLOCKS;
 #define GPIO_MCUX_LPC_IRQ(n, m)								\
 	COND_CODE_1(DT_INST_IRQ_HAS_IDX(n, m), (GPIO_MCUX_LPC_IRQ_CONNECT(n, m)), ())
 
+#define GPIO_MCUX_LPC_IRQ_DATA(n)								\
+	COND_CODE_1(DT_INST_IRQ_HAS_IDX(n, 0), (struct gpio_mcux_lpc_data *data = dev->data), ())
```

Comment:
I think it would be clearer to move this into `GPIO_MCUX_LPC_IRQ_CONNECT`

Reply:
Done